src/include/storage/recovery/recovery_manager.h

```diff
@@ -43,7 +43,20 @@ class RecoveryManager : public common::DedicatedThreadOwner {
     /**
      * Runs the recovery task. Our task only calls Recover on the log manager.
      */
-    void RunTask() override { recovery_manager_->Recover(); }
+    void RunTask() override {
+      if (recovery_manager_->enable_metric_) {
+        // Enable metrics
+        recovery_manager_->metrics_manager_->RegisterThread();
+        recovery_manager_->metrics_manager_->EnableMetric(metrics::MetricsComponent::RECOVERY);
+      }
+
+      recovery_manager_->Recover();
+
+      // Save results to metrics manager, otherwise the results are lost
+      if (recovery_manager_->enable_metric_) {
+        recovery_manager_->metrics_manager_->Aggregate();
```

Comment:
The `RecoveryManager` should not call `Aggregate`, thats the role of the `MetricsThread`. If you want to test, you can use a generous `sleep` to ensure the `MetricsThread` has enough time to wake up and Aggregate the data.